scripts/build_windows_plugins.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# 	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+set -euo pipefail
+
+# List of arguments the script can accept
+ARGUMENT_LIST=(
+  "KINESIS_PLUGIN_CLONE_URL"
+  "KINESIS_PLUGIN_TAG"
+  "KINESIS_PLUGIN_BRANCH"
+  "FIREHOSE_PLUGIN_CLONE_URL"
+  "FIREHOSE_PLUGIN_TAG"
+  "FIREHOSE_PLUGIN_BRANCH"
+  "CLOUDWATCH_PLUGIN_CLONE_URL"
+  "CLOUDWATCH_PLUGIN_TAG"
+  "CLOUDWATCH_PLUGIN_BRANCH"
+)
+
+# A variable to hold the build arguments for docker build
+PLUGIN_BUILD_ARGS=""
+
+# Method to display usage of the script
+usage() {
+  echo "Usage: $0 [--KINESIS_PLUGIN_CLONE_URL <string>] [--KINESIS_PLUGIN_TAG <string>] [--KINESIS_PLUGIN_BRANCH <string>]\
+  [--FIREHOSE_PLUGIN_CLONE_URL <string>] [--FIREHOSE_PLUGIN_TAG <string>] [--FIREHOSE_PLUGIN_BRANCH <string>]\
+  [--CLOUDWATCH_PLUGIN_CLONE_URL <string>] [--CLOUDWATCH_PLUGIN_TAG <string>] [--CLOUDWATCH_PLUGIN_BRANCH <string>]" 1>&2;
+  exit 1;
+}
+
+# Read arguments
+opts=$(getopt \
+  --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}")" \
+  --name "$(basename "$0")" \
+  --options "" \
+  -- "$@"
+)
+
+eval set -- "$opts"
+
+# Build plugin build arguments
+while [ $# -gt 0 ]
```

Comment:
I have never seen this before but not a bash expert. Why cant we just use getopt for the loop?

Reply:
So `$#` expands the number of arguments passed to the script.
So, each time when we do shift 2, we simply decrement this.

Reference: https://askubuntu.com/questions/939620/what-does-mean-in-bash

Reply:
Cool works for me.